src/include/migraphx/operators.hpp

```diff
@@ -635,17 +635,25 @@ struct as_shape
 
 struct gather
 {
-    std::size_t axis = 0;
+    mutable int axis = 0;
```

Comment:
Let not use `mutable`. We should just recompute the axis everytime we need it(it should be just a simple subtraction).